[flaky test][receiver/github] TestEvalCommits/TestMultiplePageResponse failing intermittently

### Component(s)

receiver/github

### Describe the issue you're reporting

[Failing CI/CD link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10682335912/job/29607989045?pr=34729)

Failure output:
```
=== Failed
=== FAIL: internal/scraper/githubscraper TestEvalCommits/TestMultiplePageResponse (0.00s)
    helpers_test.go:[90](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10682335912/job/29607989045?pr=34729#step:9:91)2: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/githubreceiver/internal/scraper/githubscraper/helpers_test.go:902
make[2]: *** [../../Makefile.Common:133: test-with-cover] Error 1
        	Error:      	Not equal: 
        	            	expected: 52833113
        	            	actual  : 52833114
        	Test:       	TestEvalCommits/TestMultiplePageResponse

=== FAIL: internal/scraper/githubscraper TestEvalCommits (0.01s)
```

--- a/receiver/githubreceiver/internal/scraper/githubscraper/helpers_test.go
+++ b/receiver/githubreceiver/internal/scraper/githubscraper/helpers_test.go
@@ -899,6 +899,12 @@ func TestEvalCommits(t *testing.T) {
 			assert.Equal(t, tc.expectedAge, age)
 			assert.Equal(t, tc.expectedDeletions, dels)
 			assert.Equal(t, tc.expectedAdditions, adds)
+			// assert.WithinRange(t, age, time.tc.expectedAge, time.Now())
+			if tc.expectedAge != 0 {
+				assert.WithinDuration(t, time.UnixMilli(tc.expectedAge), time.UnixMilli(age), 10*time.Second)
+			} else {
+				assert.Equal(t, tc.expectedAge, age)
+			}
 
 			if tc.expectedErr == nil {
 				assert.NoError(t, err)

